lldb/source/API/SBThread.cpp

```diff
@@ -577,7 +586,10 @@ void SBThread::StepInto(const char *target_name, uint32_t end_line,
   LLDB_INSTRUMENT_VA(this, target_name, end_line, error, stop_other_threads);
 
   std::unique_lock<std::recursive_mutex> lock;
-  ExecutionContext exe_ctx(m_opaque_sp.get(), lock);
+  Process::StopLocker stop_locker;
+  ExecutionContext exe_ctx(m_opaque_sp.get(), lock, stop_locker);
+  if (!stop_locker.IsLocked())
+    return;
```

Comment:
This one also has an error argument that you can fill in with the "process not stopped" error.